src/main/java/org/mule/extension/socket/api/client/UdpClient.java

```diff
@@ -46,8 +42,8 @@ public UdpClient(DatagramSocket socket, SocketConnectionSettings connectionSetti
    * {@inheritDoc}
    */
   @Override
-  public void write(Object data, String outputEncoding) throws IOException {
-    byte[] byteArray = getUdpAllowedByteArray(data, outputEncoding, objectSerializer);
+  public void write(InputStream data) throws IOException {
+    byte[] byteArray = getByteArray(data);
```

Comment:
why? you should be able to send the stream. this looses all the fun of streaming. verify this is not happening in other parts of the connector